float64 enu_x
float64 enu_y
float64 enu_z
int64 index
